src/data/cel/strings.rs

```diff
@@ -0,0 +1,374 @@
+                    Value::UInt(u) => *u as usize,
+                    Value::Int(i) => *i as usize,
+                    _ => Err(ExecutionError::FunctionError {
+                        function: "String.split".to_string(),
+                        message: format!(
+                            "Expects a second argument of type Integer, got `{:?}`",
+                            args[1]
+                        ),
+                    })?,
+                };
+                this.splitn(pos, sep)
+                    .map(|s| Value::String(s.to_owned().into()))
+                    .collect::<Vec<Value>>()
+            } else {
+                this.split(sep)
+                    .map(|s| Value::String(s.to_owned().into()))
+                    .collect::<Vec<Value>>()
+            };
+            Ok(list.into())
+        }
+        _ => Err(ExecutionError::FunctionError {
+            function: "String.split".to_owned(),
+            message: format!("Expects at most 2 arguments, got {args:?}"),
+        }),
+    }
+}
+
+pub fn substring(This(this): This<Arc<String>>, Arguments(args): Arguments) -> ResolveResult {
+    match args.len() {
+        count @ 1..=2 => {
+            let start = match &args[0] {
+                Value::Int(i) => *i as usize,
+                Value::UInt(u) => *u as usize,
+                _ => Err(ExecutionError::FunctionError {
+                    function: "String.substring".to_string(),
+                    message: format!(
+                        "Expects a first argument of type Integer, got `{:?}`",
+                        args[0]
+                    ),
+                })?,
+            };
+            let mut end = if count == 2 {
+                match &args[1] {
+                    Value::Int(i) => *i as usize,
+                    Value::UInt(u) => *u as usize,
+                    _ => Err(ExecutionError::FunctionError {
+                        function: "String.substring".to_string(),
+                        message: format!(
+                            "Expects a second argument of type Integer, got `{:?}`",
+                            args[0]
+                        ),
+                    })?,
+                }
+            } else {
+                this.chars().count()
+            };
+            if end < start {
+                Err(ExecutionError::FunctionError {
+                    function: "String.substring".to_string(),
+                    message: format!("Can't have end be before the start: `{end} < {start}"),
+                })?
+            }
+            end -= start;
+            Ok(this
+                .chars()
+                .skip(start)
+                .take_while(|_| {
+                    if let Some(v) = end.checked_sub(1) {
+                        end = v;
+                        true
+                    } else {
+                        false
+                    }
+                })
+                .collect::<String>()
+                .into())
```

Comment:
Maybe I'm missing a case but couldn't we just avoid this `take_while`? although maybe this is related to what we

```suggestion
            Ok(this
                .chars()
                .skip(start)
                .take(end - start)
                .collect::<String>()
                .into())
```